### PR TITLE
Camera Controller (Provides default navigation functionality over orthographic camera).

### PR DESCRIPTION
**Pull request by TejasJD**
- Wrapper around the Orthographic Camera that provides default inputs for navigating the camera around the 2D scene.
- Configurable rotation and translation speeds for the Camera Controller.
- Takes care of the geometry without distorting anything, upon resizing of the window and during zoom in and out.

### CODE REVIEW

File: HorizonEngine/src/HorizonEngine/App.cpp [hunk 1]
```diff
@@ -14,8 +14,8 @@ namespace Hzn
 		registerComponents();
 		/*HZN_CORE_ASSERT(false, "application already initialized");*/
 		m_Instance = this;
-		m_Input = std::unique_ptr<Input>(Input::create());
-		m_Window = std::unique_ptr<Window>(Window::create(800, 600, "HorizonEngine"));
+		m_Input = Input::create();
+		m_Window = Window::create(800, 600, "HorizonEngine");
```
Comment (Minghui-Gao) [inline HorizonEngine/src/HorizonEngine/App.cpp:18]: Why did you cancel the use of unique_ptr here?

File: HorizonEngine/Platform/OpenGL/GLWindow.h [hunk 2]
```diff
@@ -42,6 +42,6 @@ namespace Hzn
 		GLFWwindow* m_Window;
 		bool m_Vsync = false;
 
-		std::unique_ptr<RenderContext> m_Context;
+		std::shared_ptr<RenderContext> m_Context;
```
Comment (Minghui-Gao) [inline HorizonEngine/Platform/OpenGL/GLWindow.h:45]: Why did you use shared_ptr instead of unique_ptr here

Comment (TejasJD) [reply HorizonEngine/Platform/OpenGL/GLWindow.h:45]: Hi,
For now, the create methods for these classes return a reference counted pointer (shared_ptr), so I decided to change the fields in the App class to receive those.

File: HorizonEngine/src/HorizonEngine/App.cpp (hunk 1 above)
Comment (TejasJD) [reply HorizonEngine/src/HorizonEngine/App.cpp:18]: The create methods return shared pointers so I receive the pointer to those instances as shared pointers.

File: HorizonEngine/Platform/OpenGL/GLWindow.h (hunk 2 above)
Comment (Minghui-Gao) [reply HorizonEngine/Platform/OpenGL/GLWindow.h:45]: Oh okay.Thank you for resolving my doubts

**Review by BozhidarVelikov — APPROVED** [review]
It's really good that the camera is not distorting the objects on the screen. Good job!

**Review by prajakta-kulkarni-git — APPROVED** [review]
Looks good.

**Pull request merged**